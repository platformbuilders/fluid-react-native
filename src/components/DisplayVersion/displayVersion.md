
# Button ðŸš€

### Simple Button Example:

<!-- STORY -->

### Source Code
<!-- SOURCE -->

<!-- PROPS -->

<!-- STORY HIDE START -->

content here won't be shown in stories

<!-- STORY HIDE END -->

