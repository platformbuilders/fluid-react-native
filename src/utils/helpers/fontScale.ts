import { RFValue } from 'react-native-responsive-fontsize';

export const scaledFontSize = (
  size: number | string,
  baselineFactor?: 280,
): number => {
  const value = typeof size === 'string' ? parseInt(size, 10) : size;
  return RFValue(value, baselineFactor);
};
