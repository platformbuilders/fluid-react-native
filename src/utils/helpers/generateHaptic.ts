import Haptic, { HapticFeedbackType } from 'react-native-haptic';
import { isIOS } from './isIOS';

export const generateHaptic = (haptic: HapticFeedbackType = 'impact'): void => {
  if (isIOS()) {
    Haptic.generate(haptic);
  }
};
