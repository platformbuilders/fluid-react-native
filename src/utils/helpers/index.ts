export {
  currencyParser,
  parseToThousands,
  removeWhiteSpaces,
  toOnlyNumbers,
  getTheme,
  ifStyle,
  switchStyle,
} from '@platformbuilders/helpers';
export * from './generateHaptic';
export * from './getByTypographyVariant';
export * from './fontScale';
export * from './shadow';
