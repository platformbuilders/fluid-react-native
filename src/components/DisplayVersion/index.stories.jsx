import React from 'react';
import { storiesOf } from '@storybook/react-native';
import DisplayVersion from './index';
import displayVersionMarkdown from './displayVersion.md';

storiesOf('DisplayVersion', module)
  .addParameters({
    readme: {
      content: displayVersionMarkdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => <DisplayVersion appVersion="1.0" buildVersion="1" />);
