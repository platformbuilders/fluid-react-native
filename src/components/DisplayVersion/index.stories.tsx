import type { Meta } from '@storybook/react';
import DisplayVersion from '.';

const meta: Meta<typeof DisplayVersion> = {
  title: 'DisplayVersion',
  component: DisplayVersion,
  args: {
    appVersion: '1.4.5',
    buildVersion: '9090',
  },
  parameters: {
    notes:
      'Esse componente tem como função auxiliar na padronização de exibição da versão do aplicativo.',
  },
};

export default meta;

export const Default = {};
