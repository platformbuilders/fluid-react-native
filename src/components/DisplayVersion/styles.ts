import styled from 'styled-components/native';

import { getTheme } from '../../utils/helpers';

import Typography from '../Typography';

const primaryMain = getTheme('primary.main');
const smallSpacing = getTheme('smallSpacing');

export const Text = styled(Typography).attrs({
  variant: 'subhead',
})`
  color: ${primaryMain};
  text-align: center;
  margin-vertical: ${smallSpacing};
`;
