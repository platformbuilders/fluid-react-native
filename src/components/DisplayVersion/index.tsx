/* eslint-disable react/jsx-one-expression-per-line */
import React, { FC } from 'react';
import { Text } from './styles';

type Props = {
  appVersion?: string;
  buildVersion?: string;
};

const DisplayVersion: FC<Props> = ({ appVersion, buildVersion }) => (
  <Text accessibility="display-version">
    Vers√£o {appVersion}
    {!!buildVersion && '/'}
    {buildVersion}
  </Text>
);

export default DisplayVersion;
