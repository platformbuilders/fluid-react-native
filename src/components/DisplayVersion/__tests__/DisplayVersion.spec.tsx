import 'jest';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import DisplayVersion from '..';
import { theme } from '../../../test/helpers';

describe('<DisplayVersion />', () => {
  it('should render and match snapshot for default props', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <DisplayVersion appVersion="1.0" buildVersion="1" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
