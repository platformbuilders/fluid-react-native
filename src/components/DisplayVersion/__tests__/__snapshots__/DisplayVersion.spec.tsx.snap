// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayVersion /> should render and match snapshot for default props 1`] = `
<Text
  accessibility="display-version"
  accessibilityLabel="display-version"
  style={
    Array [
      Object {
        "color": "#212121",
        "fontSize": 29,
        "lineHeight": 39,
      },
      Object {
        "color": "#C6B09E",
        "marginVertical": 18.857142857142858,
        "textAlign": "center",
      },
    ]
  }
  testID="display-version"
  variant="subhead"
>
  Vers√£o 
  1.0
  /
  1
</Text>
`;
